### PR TITLE
luci-app-dawn: add luci app for dawn

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Dawn is a decentralized WiFi controller.
Just install dawn and the APs will find each other via umdns. They periodically exchange information about connected clients, wireless statistics and other needed information. With that, the daemon load balances clients between different APs through association control.

Luci-app-dawn is the graphical user interface.
It allows to:
- Configure dawn
- View Wireless Network Overview
- View Hearing Map

The hearing map is the list of all probe requests seen from a client from all APs that are running the controller.

networkoverview

hearingmapoverview

configurationdawn

https://github.com/openwrt/packages/pull/11648

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
Before merging, please wait 1 sec. :)

**Comment by PolynomialDivision** [comment]
> Before merging, please wait 1 sec. :)

Finished. :) Please merge. :P

File: applications/luci-app-dawn/Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+include $(TOPDIR)/rules.mk
+
+LUCI_TITLE:=LuCI support for DAWN
+LUCI_DEPENDS:=+luci +dawn +luci-lib-json +luci-lib-httpclient +luci-compat
```
Comment (hnyman) [inline applications/luci-app-dawn/Makefile]: Dependency on the "luci" meta-package sounds wrong.

Comment (PolynomialDivision) [reply applications/luci-app-dawn/Makefile]: Oh, okay, I will delete the `+luci`. I had this in my own repository and I think this was needed. I will delete that.

**Comment by PolynomialDivision** [comment]
@hnyman I removed all dependencies and tested again! Works! :) Could u merge?

**Comment by hnyman** [comment]
I could merge it otherwise, but I doubt that your real name is Polynomialdivision 
Please fix the commit message to have the signed off with your real name.

**Comment by PolynomialDivision** [comment]
> I could merge it otherwise, but I doubt that your real name is Polynomialdivision
> Please fix the commit message to have the signed off with your real name.

I did it. But now check fails. :/

**Comment by hnyman** [comment]
I will merge it.
The next step would be to change it for client-side JS, instead of lua.

**Comment by castillofrancodamian** [comment]
And create the template to translate.

**Comment by PolynomialDivision** [comment]
The package is still not appearing on the web server? Is something wrong? I can not find any faillogs...

**Comment by hnyman** [comment]
It is built for about 50 different targets, one by one. 
http://buildbot.openwrt.org/master/packages/grid
http://buildbot.openwrt.org/master/packages/one_line_per_build

image

**Comment by PolynomialDivision** [comment]
Ah thanks! I thought every packet would be build around 0:00... :)

**Comment by castillofrancodamian** [comment]
The translation template is missing in the application directory. I couldn't create it with i18n-scan.pl/i18n-update.pl (there are other apps that I can't get to work either).

**Comment by PolynomialDivision** [comment]
> The translation template is missing in the application directory. I couldn't create it with i18n-scan.pl/i18n-update.pl (there are other apps that I can't get to work either).

I did not create any translation template? Could u post the error? I'm very thankful for PRs. :)

**Comment by hnyman** [comment]
The template would make no difference, as most of strings in the app have not been marked as translatable.  (no "translate" wrappers on strings)

**Comment by hoangminh88** [comment]
How install dawn & luci-app-dawn in Openwrt 19.07.3 stable?

**Comment by Xtreme512** [comment]
> How install dawn & luci-app-dawn in Openwrt 19.07.3 stable?

exactly... I want to try it out for if it helps my 802.11k and .11v problems, posted also in OpenWrt forums. we need a stable release.

**Pull request merged**